### PR TITLE
poetryPlugins.poetry-plugin-poeblix: init at 0.10.0

### PR DESCRIPTION
**Pull request by hennk**
## Description of changes

https://github.com/spoorn/poeblix

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/tools/package-management/poetry/plugins/poetry-plugin-poeblix.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    sed -i '/poetry =/d' pyproject.toml
+  '';
+
+  nativeBuildInputs = [
+    poetry-core
+  ];
+
+  # infinite recursion with poetry
+  doCheck = false;
+  pythonImportsCheck = [];
```
Comment (SuperSandro2000) [inline pkgs/tools/package-management/poetry/plugins/poetry-plugin-poeblix.nix:29]: Why not do an import check?

File: pkgs/tools/package-management/poetry/plugins/poetry-plugin-poeblix.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, poetry-core
+}:
+
+buildPythonPackage rec {
+  pname = "poetry-plugin-poeblix";
+  version = "0.10.0";
+  format = "pyproject";
```
Comment (SuperSandro2000) [inline pkgs/tools/package-management/poetry/plugins/poetry-plugin-poeblix.nix:10]:
```suggestion
  pyproject = true;
```

File: pkgs/tools/package-management/poetry/plugins/poetry-plugin-poeblix.nix (hunk 1 above)
Comment (hennk) [reply pkgs/tools/package-management/poetry/plugins/poetry-plugin-poeblix.nix]: To be honest, I basically adapted the package of one of the existing poetry plugins. I just checked, and it builds fine without deactivating the checks, so I removed those lines

File: pkgs/tools/package-management/poetry/plugins/poetry-plugin-poeblix.nix (hunk 2 above)
Comment (hennk) [reply pkgs/tools/package-management/poetry/plugins/poetry-plugin-poeblix.nix]: Changed, thanks

File: pkgs/tools/package-management/poetry/default.nix [hunk 3]
```diff
@@ -30,6 +30,7 @@ let
     poetry-audit-plugin = callPackage ./plugins/poetry-audit-plugin.nix { };
     poetry-plugin-export = callPackage ./plugins/poetry-plugin-export.nix { };
     poetry-plugin-up = callPackage ./plugins/poetry-plugin-up.nix { };
+    poetry-plugin-poeblix = callPackage ./plugins/poetry-plugin-poeblix.nix { };
```
Comment (dotlambda) [inline pkgs/tools/package-management/poetry/default.nix:33]:
```suggestion
    poeblix = callPackage ./plugins/poeblix.nix { };
```
and even before that it wasn't sorted alphabetically.

**Review by dotlambda — COMMENTED** [review]
Why weren't the poetry maintainers pinged?

**Comment by hennk** [comment]
> Why weren't the poetry maintainers pinged?

@dotlambda Sorry, this was my first PR, I'll keep that in mind in the future. Should I open a new PR for the changes you suggest?

**Comment by dotlambda** [comment]
I gave this a try since you don't seem to be making progress:
```diff
diff --git a/pkgs/tools/package-management/poetry/default.nix b/pkgs/tools/package-management/poetry/default.nix
index 415a2c0274d3..01745041f96b 100644
--- a/pkgs/tools/package-management/poetry/default.nix
+++ b/pkgs/tools/package-management/poetry/default.nix
@@ -27,10 +27,10 @@ let
   });
 
   plugins = ps: with ps; {
+    poeblix = callPackage ./plugins/poeblix.nix { };
     poetry-audit-plugin = callPackage ./plugins/poetry-audit-plugin.nix { };
     poetry-plugin-export = callPackage ./plugins/poetry-plugin-export.nix { };
     poetry-plugin-up = callPackage ./plugins/poetry-plugin-up.nix { };
-    poetry-plugin-poeblix = callPackage ./plugins/poetry-plugin-poeblix.nix { };
   };
 
   # selector is a function mapping pythonPackages to a list of plugins
diff --git a/pkgs/tools/package-management/poetry/plugins/poeblix.nix b/pkgs/tools/package-management/poetry/plugins/poeblix.nix
index 6e8cad3ef953..c73ca64d3a40 100644
--- a/pkgs/tools/package-management/poetry/plugins/poeblix.nix
+++ b/pkgs/tools/package-management/poetry/plugins/poeblix.nix
@@ -2,6 +2,9 @@
 , buildPythonPackage
 , fetchFromGitHub
 , poetry-core
+, poetry
+, pytestCheckHook
+, python
 }:
 
 buildPythonPackage rec {
@@ -17,19 +20,32 @@ buildPythonPackage rec {
   };
 
   postPatch = ''
-    sed -i '/poetry =/d' pyproject.toml
+    sed -i "/^addopts/d" pyproject.toml
   '';
 
-  nativeBuildInputs = [
+  build-system = [
     poetry-core
   ];
 
-  doCheck = false;
-  pythonImportsCheck = ["poeblix"];
+  buildInputs = [
+    poetry
+  ];
+
+  pythonImportsCheck = [ "poeblix" ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+  ];
+
+  pytestFlagsArray = [
+    "--pspec" "test_walkthrough.py"
+  ];
+
+  #doCheck = with python; isPy37 || isPy39;
 
   meta = with lib; {
     changelog = "https://github.com/spoorn/poeblix/releases/tag/${src.rev}";
-    description = "Poetry Plugin that adds various features that extend the poetry command such as building wheel files with locked dependencies, and validations of wheel/docker containers";
+    description = "Poetry plugin for building wheel files with locked dependencies and validating wheel/docker containers";
     license = licenses.mit;
     homepage = "https://github.com/spoorn/poeblix";
     maintainers = with maintainers; [ hennk ];
```
It didn't work so let's revert this change for now.

**Comment by hennk** [comment]
The poeblix tests assume that the plugin is installed in poetry, already (see https://github.com/spoorn/poeblix/blob/main/README.md#development , the `./devtool bootstrap` line installs poeblix into the currently installed poetry instance).

I don't know how to do this inside the check phase, maybe you have an idea?

**Comment by hennk** [comment]
I started a PR at https://github.com/NixOS/nixpkgs/pull/300432 to incorporate the changes you requested @dotlambda checks are still disabled, though, due to the problem mentioned above

**Pull request merged**